evaluate/task/write-test.go

```diff
@@ -46,44 +53,87 @@ func (t *WriteTests) Run(ctx evaltask.Context) (repositoryAssessment map[evaltas
 
 	modelAssessment := metrics.NewAssessments()
 	withSymflowerFixAssessment := metrics.NewAssessments()
+	withSymflowerTemplateAssessment := metrics.NewAssessments()
+	withSymflowerTemplateAndFixAssessment := metrics.NewAssessments()
 
 	maximumReachableFiles := uint64(len(filePaths))
 	modelAssessment[metrics.AssessmentKeyFilesExecutedMaximumReachable] = maximumReachableFiles
 	withSymflowerFixAssessment[metrics.AssessmentKeyFilesExecutedMaximumReachable] = maximumReachableFiles
+	withSymflowerTemplateAssessment[metrics.AssessmentKeyFilesExecutedMaximumReachable] = maximumReachableFiles
+	withSymflowerTemplateAndFixAssessment[metrics.AssessmentKeyFilesExecutedMaximumReachable] = maximumReachableFiles
 
 	for _, filePath := range filePaths {
+		// Handle this task case without a template.
 		if err := ctx.Repository.Reset(ctx.Logger); err != nil {
 			ctx.Logger.Panicf("ERROR: unable to reset temporary repository path: %s", err)
 		}
 
-		modelAssessmentFile, withSymflowerFixAssessmentFile, ps, err := runModelAndSymflowerFix(ctx, taskLogger, modelCapability, dataPath, filePath)
+		modelAssessmentFile, withSymflowerFixAssessmentFile, ps, err := runModelAndSymflowerFix(ctx, taskLogger, modelCapability, dataPath, filePath, &ArgumentsWriteTest{})
 		problems = append(problems, ps...)
 		if err != nil {
 			return nil, problems, err
 		}
 		modelAssessment.Add(modelAssessmentFile)
 		withSymflowerFixAssessment.Add(withSymflowerFixAssessmentFile)
+
+		if !(ctx.Language.ID() == "golang" || ctx.Language.ID() == "java") { // Symflower templates currently only exist for Go and Java.
```

Comment:
This should be part of the clean up. the tasks, should not care about the language. the language adapter itself says if it implements something or not. Tasks should be written with interfaces.